react/Shelf.tsx

```diff
@@ -0,0 +1,163 @@
+    )
+  }
+  return (
+    <ProductSummaryListWithQuery {...props}>
+      {children}
+    </ProductSummaryListWithQuery>
+  )
+}
+
+Shelf.getSchema = () => ({
+  title: 'admin/editor.shelf.title',
+  description: 'admin/editor.shelf.description',
+  type: 'object',
+  properties: {
+    category: {
+      title: 'admin/editor.productSummaryList.category.title',
+      description: 'admin/editor.productSummaryList.category.description',
+      type: 'string',
+      isLayout: false,
+    },
+    specificationFilters: {
+      title: 'admin/editor.productSummaryList.specificationFilters.title',
+      type: 'array',
+      items: {
+        title:
+          'admin/editor.productSummaryList.specificationFilters.item.title',
+        type: 'object',
+        properties: {
+          id: {
+            type: 'string',
+            title:
+              'admin/editor.productSummaryList.specificationFilters.item.id.title',
+          },
+          value: {
+            type: 'string',
+            title:
+              'admin/editor.productSummaryList.specificationFilters.item.value.title',
+          },
+        },
+      },
+    },
+    collection: {
+      title: 'admin/editor.productSummaryList.collection.title',
+      type: 'string',
+      isLayout: false,
+    },
+    orderBy: {
+      title: 'admin/editor.productSummaryList.orderBy.title',
+      type: 'string',
+      enum: getOrdinationProp('value'),
+      enumNames: getOrdinationProp('name'),
+      default: ORDER_BY_OPTIONS[1].value,
+      isLayout: false,
+    },
+    hideUnavailableItems: {
+      title: 'admin/editor.productSummaryList.hideUnavailableItems',
+      type: 'boolean',
+      default: false,
+      isLayout: false,
+    },
+    maxItems: {
+      title: 'admin/editor.productSummaryList.maxItems.title',
+      type: 'number',
+      isLayout: false,
+      default: 10,
+    },
+    skusFilter: {
+      title: 'admin/editor.productSummaryList.skusFilter.title',
+      description: 'admin/editor.productSummaryList.skusFilter.description',
+      type: 'string',
+      default: 'ALL_AVAILABLE',
+      enum: ['ALL_AVAILABLE', 'ALL', 'FIRST_AVAILABLE'],
+      enumNames: [
+        'admin/editor.productSummaryList.skusFilter.all-available',
+        'admin/editor.productSummaryList.skusFilter.none',
+        'admin/editor.productSummaryList.skusFilter.first-available',
+      ],
+    },
+    installmentCriteria: {
+      title: 'admin/editor.productSummaryList.installmentCriteria.title',
+      description:
+        'admin/editor.productSummaryList.installmentCriteria.description',
+      type: 'string',
+      default: 'MAX_WITHOUT_INTEREST',
+      enum: ['MAX_WITHOUT_INTEREST', 'MAX_WITH_INTEREST'],
+      enumNames: [
+        'admin/editor.productSummaryList.installmentCriteria.max-without-interest',
+        'admin/editor.productSummaryList.installmentCriteria.max-with-interest',
+      ],
+    },
+  },
+})
```

Comment:
You should do `Shelf.schema = { title: 'my-title' }` and add the schema to the `contentSchemas.json` file